### PR TITLE
use cobra command in edgesite and edgecore

### LINKED ISSUE #971: Use cobra command and Remove init to use explicit call
Now in some components like cloud/edge core, all used init to load config and register sub modules. And even if we want to only print the version of component, the init func still be called and print lots of logs, so we need to `Remove init to use explicit call` and `Use cobra command`.

The task lists:

- [x] Use cobra command and Remove init in cloud part #988 
- [ ] Use cobra command and Remove init in edge core
- [ ] Use cobra command and Remove init in edgesite
- [ ] TBD

/cc @rohitsardesai83 @sids-b @kevin-wangzefeng @qizha @edisonxiang @Baoqiang-Zhang @kadisi @chendave @subpathdev

### PR DESCRIPTION
**Pull request by kexun**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
> /kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #971 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: edge/cmd/edgecore/app/server.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/kubeedge/kubeedge/edge/pkg/eventbus"
+	"github.com/kubeedge/kubeedge/edge/pkg/metamanager"
+	"github.com/kubeedge/kubeedge/edge/pkg/servicebus"
+)
+
+// NewEdgeCoreCommand create edgecore cmd
+func NewEdgeCoreCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "edgecore",
+		Long: `Edgecore is the core edge part of KubeEdge, which contains three modules: devicetwin, edged,
```
Comment (fisherxu) [inline edge/cmd/edgecore/app/server.go]: s/three/six/

File: edge/cmd/edgecore/app/server.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/kubeedge/kubeedge/edge/pkg/metamanager"
+	"github.com/kubeedge/kubeedge/edge/pkg/servicebus"
+)
+
+// NewEdgeCoreCommand create edgecore cmd
+func NewEdgeCoreCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "edgecore",
+		Long: `Edgecore is the core edge part of KubeEdge, which contains three modules: devicetwin, edged, 
+edgehub, eventbus, metamanager, and servicebus. Edgecore is an agent that runs on edge nodes and
```
Comment (fisherxu) [inline edge/cmd/edgecore/app/server.go]: Edged

File: edge/cmd/edgecore/app/server.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+// NewEdgeCoreCommand create edgecore cmd
+func NewEdgeCoreCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "edgecore",
+		Long: `Edgecore is the core edge part of KubeEdge, which contains three modules: devicetwin, edged, 
+edgehub, eventbus, metamanager, and servicebus. Edgecore is an agent that runs on edge nodes and 
+manages containerized applications and devices. Edgehub is a web socket client responsible for 
+interacting with Cloud Service for the edge computing (like Edge Controller as in the KubeEdge 
+Architecture). This includes syncing cloud-side resource updates to the edge, and reporting 
+edge-side host and device status changes to the cloud.`,
```
Comment (fisherxu) [inline edge/cmd/edgecore/app/server.go]: Need contain other modules here

File: edge/cmd/edgecore/edgecore.go [hunk 4]
```diff
@@ -1,8 +1,9 @@
 package main
 
 import (
-	"github.com/kubeedge/beehive/pkg/core"
-	"github.com/kubeedge/kubeedge/edge/pkg/common/dbm"
+	"os"
+
+	"github.com/kubeedge/kubeedge/edge/cmd/edgecore/app"
 	_ "github.com/kubeedge/kubeedge/edge/pkg/devicetwin"
```
Comment (fisherxu) [inline edge/cmd/edgecore/edgecore.go]: remove this unused import

File: edge/pkg/devicetwin/devicetwin.go [hunk 5]
```diff
@@ -14,7 +14,8 @@ type DeviceTwin struct {
 	dtcontroller *DTController
 }
 
-func init() {
+// Register register device
```
Comment (fisherxu) [inline edge/pkg/devicetwin/devicetwin.go]: devicetwin?

**Comment by fisherxu** [comment]
thanks for implement, seems UT need to be fixed :)

File: edge/cmd/edgecore/app/server.go (hunk 2 above)
Comment (kexun) [reply edge/cmd/edgecore/app/server.go]: fixed

File: edge/cmd/edgecore/app/server.go (hunk 1 above)
Comment (kexun) [reply edge/cmd/edgecore/app/server.go]: fixed

File: edge/cmd/edgecore/app/server.go (hunk 3 above)
Comment (kexun) [reply edge/cmd/edgecore/app/server.go]: fixed

File: edge/cmd/edgecore/edgecore.go (hunk 4 above)
Comment (kexun) [reply edge/cmd/edgecore/edgecore.go]: removed

File: edge/pkg/devicetwin/devicetwin.go (hunk 5 above)
Comment (kexun) [reply edge/pkg/devicetwin/devicetwin.go]: fixed

File: edge/cmd/edgecore/app/server.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+func NewEdgeCoreCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "edgecore",
+		Long: `Edgecore is the core edge part of KubeEdge, which contains six modules: devicetwin, edged, 
+edgehub, eventbus, metamanager, and servicebus. Edged is an agent that runs on edge nodes and 
+manages containerized applications and devices. Edgehub is a web socket client responsible for 
+interacting with Cloud Service for the edge computing (like Edge Controller as in the KubeEdge 
+Architecture). This includes syncing cloud-side resource updates to the edge, and reporting 
+edge-side host and device status changes to the cloud. EventBus is a MQTT client to interact with MQTT 
+servers (mosquito), offering publish and subscribe capabilities to other components.  ServiceBus is
```
Comment (fisherxu) [inline edge/cmd/edgecore/app/server.go]: Duplicate spaces here, and please Corresponding to the module order above, `devicetwin, edged, 
edgehub, eventbus, metamanager, and servicebus.`

File: edge/pkg/edgehub/controller_test.go [hunk 7]
```diff
@@ -54,6 +59,12 @@ var mockAdapter *edgehub.MockAdapter
 
 //init() starts the test server and generates test certificates for testing
 func init() {
+	devicetwin.Register()
+	//edged.Register()
```
Comment (fisherxu) [inline edge/pkg/edgehub/controller_test.go]: Removed unused line

File: edge/cmd/edgecore/edgecore.go [hunk 8]
```diff
@@ -1,19 +1,17 @@
...
-	_ "github.com/kubeedge/kubeedge/edge/pkg/devicetwin"
-	_ "github.com/kubeedge/kubeedge/edge/pkg/edged"
-	_ "github.com/kubeedge/kubeedge/edge/pkg/edgehub"
-	_ "github.com/kubeedge/kubeedge/edge/pkg/eventbus"
-	_ "github.com/kubeedge/kubeedge/edge/pkg/metamanager"
-	_ "github.com/kubeedge/kubeedge/edge/pkg/servicebus"
+	"os"
+
+	"github.com/kubeedge/kubeedge/edge/cmd/edgecore/app"
 	_ "github.com/kubeedge/kubeedge/edge/test"
```
Comment (fisherxu) [inline edge/cmd/edgecore/edgecore.go]: @edisonxiang @sids-b @sujithsimon22 Could you please help check whether the test need import here? Thanks! :)

Comment (sids-b) [reply edge/cmd/edgecore/edgecore.go]: The test should not be imported here at it is required only for integration test and e2e tests. It should be removed from the actual binary and should be add via code in the tests if required.
@edisonxiang @pavan187

**Comment by sids-b** [comment]
Thanks @kexun 

/approve

**Pull request merged**